WordPress/WordPressShareExtension/ShareTagsPickerViewController.swift

```diff
@@ -0,0 +1,467 @@
+    fileprivate let textView = UITextView()
+    fileprivate let textViewContainer = UIView()
+    fileprivate let tableView = UITableView(frame: .zero, style: .grouped)
+    fileprivate var dataSource: PostTagPickerDataSource = LoadingDataSource() {
+        didSet {
+            tableView.dataSource = dataSource
+            reloadTableData()
+        }
+    }
+
+    // MARK: - Initializers
+
+    init(siteID: Int, tags: String?) {
+        self.originalTags = tags?.arrayOfTags() ?? [String()]
+        self.siteID = siteID
+        super.init(nibName: nil, bundle: nil)
+        textView.text = normalizeInitialTags(tags: originalTags)
+        textViewDidChange(textView)
+    }
+
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    // MARK: - View Lifecycle
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        // Initialize Interface
+        setupNavigationBar()
+        setupTableView()
+        setupTextView()
+        setupConstraints()
+        keyboardObserver.tableView = tableView
+    }
+
+    override func viewDidAppear(_ animated: Bool) {
+        super.viewDidAppear(animated)
+        updateSuggestions()
+        textView.becomeFirstResponder()
+        loadTags()
+    }
+
+    override func viewWillTransition(to size: CGSize, with coordinator: UIViewControllerTransitionCoordinator) {
+        super.viewWillTransition(to: size, with: coordinator)
+
+        coordinator.animate(alongsideTransition: { _ in
+            self.stopEditing()
+        })
+    }
+
+    // MARK: - Setup Helpers
+
+    fileprivate func setupNavigationBar() {
+        self.navigationItem.hidesBackButton = true
+        navigationItem.leftBarButtonItem = cancelButton
+        navigationItem.rightBarButtonItem = applyButton
+    }
+
+    fileprivate func setupTableView() {
+        WPStyleGuide.configureColors(for: view, andTableView: tableView)
+        tableView.register(UITableViewCell.self, forCellReuseIdentifier: SuggestionsDataSource.cellIdentifier)
+        tableView.register(LoadingDataSource.Cell.self, forCellReuseIdentifier: LoadingDataSource.cellIdentifier)
+        tableView.register(FailureDataSource.Cell.self, forCellReuseIdentifier: FailureDataSource.cellIdentifier)
+        tableView.delegate = self
+        tableView.dataSource = dataSource
+        tableView.tableHeaderView = UIView(frame: CGRect(x: 0, y: 0, width: 0, height: CGFloat.leastNonzeroMagnitude))
+        reloadTableData()
+    }
+
+    fileprivate func setupTextView() {
+        textView.delegate = self
+        textView.autocorrectionType = .yes
+        textView.autocapitalizationType = .none
+        textView.font = WPStyleGuide.tableviewTextFont()
+        textView.textColor = WPStyleGuide.darkGrey()
+        textView.isScrollEnabled = false
+        textView.textContainer.lineFragmentPadding = 0
+        textView.textContainerInset = UIEdgeInsets(top: 11, left: 0, bottom: 11, right: 0)
```

Comment:
Could we have a constants enum for this and the textview container leading/trailing padding below?